src/main/mondrian/rolap/RolapNativeSet.java

```diff
@@ -260,6 +265,32 @@ protected TupleList executeList(final SqlTupleReader tr) {
                         dataSource, partialResult, newPartialResult);
             }
 
+            // Did not get as many members as expected - try to complete using
+            // less constraints
+            if (completeWithNullValues && result.size() < maxRows) {
+                RolapLevel l = args[0].getLevel();
+                List<RolapMember> list = new ArrayList<RolapMember>();
+                for (List<Member> lm : result) {
+                    for (Member m : lm) {
+                        list.add((RolapMember) m);
+                    }
+                }
+                SqlTupleReader str = new SqlTupleReader(
+                    new MemberExcludeConstraint(
+                        list, l,
+                        constraint instanceof SetConstraint
+                        ? (SetConstraint) constraint : null));
+                str.setAllowHints(false);
```

Comment:
@mkambol, finally I got what you mean :)
In Pedro's changes the flag was used here: https://github.com/akhayrutdinov/mondrian/blob/MONDRIAN-1272-pedro/src/main/mondrian/rolap/SqlTupleReader.java#L927

I missed this and suprisingly it didn't affect test executions's results. I should either use it or remove. Will do it tomorrow. Thanks for noticing it!